src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -0,0 +1,306 @@
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param descriptor: The proto descriptor
+
+        """
+
+        # Extract the options from the descriptor, and store it
+        # in the dictionary.
+        options = MessageToDict(
+            descriptor.GetOptions(), preserving_proto_field_name=True
+        )
+
+        try:
+            min_max = options["[TbotsProto.bounds]"]
+        except KeyError:
+            raise KeyError("{} missing ParameterRangeOptions".format(key))
+
+        return {
+            "name": key,
+            "type": "slider",
+            "value": value,
+            "default": value,
+            "limits": (int(min_max["min_int_value"]), int(min_max["max_int_value"])),
+            "step": 1,
+        }
+
+    @staticmethod
+    def __create_double_parameter(key, value, descriptor):
+        """Converts a double field of a proto to a SliderParameter with
+        the min/max bounds set according to the provided ParameterRangeOptions
+        min/vax options.
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param descriptor: The proto descriptor
+
+        """
+
+        # Extract the options from the descriptor, and store it
+        # in the dictionary.
+        options = MessageToDict(
+            descriptor.GetOptions(), preserving_proto_field_name=True
+        )
+
+        try:
+            min_max = options["[TbotsProto.bounds]"]
+        except KeyError:
+            raise KeyError("{} missing ParameterRangeOptions".format(key))
+
+        return {
+            "name": key,
+            "type": "slider",
+            "value": value,
+            "default": value,
+            "limits": (min_max["min_double_value"], min_max["max_double_value"],),
+            "step": 0.01,
+        }
+
+    @staticmethod
+    def __create_enum_parameter(key, value, descriptor):
+        """Converts an enum field in a protobuf to a ListParameter. Uses
+        the options to lookup all possible enum values and provides them
+        as a dropdown option.
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param descriptor: The proto descriptor
+
+        """
+        options = []
+
+        for enum_desc in descriptor.enum_type.values:
+            options.append(enum_desc.name)
+
+        # The list index is indexed from 1
+        current_enum_index = value - 1
+
+        return parametertree.parameterTypes.ListParameter(
+            name=key,
+            default=None,
+            value=descriptor.enum_type.values[current_enum_index].name,
+            limits=options,
+        )
+
+    @staticmethod
+    def __create_bool_parameter(key, value, descriptor):
+        """Convert a bool field in proto to a BoolParameter
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param _: The proto descriptor, unused for bool
```

Comment:
nit: I think here the `descriptor` argument should be `_`, or the comment `_` be `descriptor` so its slightly more clear

Reply:
fixed